src/api/app/models/project.rb

```diff
@@ -806,18 +806,22 @@ def can_be_released_to_project?(target_project)
   end
 
   def exists_package?(name, opts = {})
-    Rails.cache.fetch(['exists_package', cache_key, name, opts]) do
-      if opts[:follow_project_links]
-        pkg = find_package(name)
-      else
-        pkg = packages.find_by_name(name)
-      end
-      if pkg.nil?
-        # local project, but package may be in a linked remote one
-        opts[:allow_remote_packages] && Package.exists_on_backend?(name, self.name)
-      else # if we could fetch the project, the package is fine accesswise
-        true
+    if opts[:follow_project_links]
+      # Look for any package with name in all our linked projects
+      pkg = Package.find_by(project: expand_linking_to, name: name)
+      # FIXME: Mimics the behavior of find_package below, but why is accesss only
+      # checked in the case of follow_project_links?
```

Comment:
Maybe because  ``Project.find_by_name`` is overwritten in line 462 and checks the access? Anyway, I guess it is a bug and I would drop the comment and check for access instead.